gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -153,7 +161,7 @@ public void shutdownNow() {
 
   @Override
   public boolean awaitTermination(long duration, TimeUnit unit) throws InterruptedException {
-    return executor.awaitTermination(duration, unit);
+    return countDownLatch.await(duration, unit);
```

Comment:
We are only waiting for the current work to be completed, not waiting for the cancellation of future here, as the definition of `awaitTermination` is `Blocks until all work has completed execution after a shutdown request`